main.js

```diff
@@ -178,4 +178,319 @@ const data = [
+        country: 'Greece',
+    },
+    {
+        name: 'Cityden Apartments',
+        city: 'Amsterdam',
+        country: 'Netherlands',
+    },
+    {
+        name: 'Mandarin Oriental',
+        city: 'Miami',
+        country: 'USA',
+    },
+    {
+        name: 'Concept Terrace Hotel',
+        city: 'Rome',
+        country: 'Italy',
+    },
+    {
+        name: 'Ponta Mar Hotel',
+        city: 'Fortaleza',
+        country: 'Brazil',
+    },
+    {
+        name: 'Four Seasons Hotel',
+        city: 'Sydney',
+        country: 'Australia',
+    },
+    {
+        name: 'Le Meridien',
+        city: 'Nice',
+        country: 'France',
+    },
+    {
+        name: 'Apart Neptun',
+        city: 'Gdansk',
+        country: 'Poland',
+    },
+    {
+        name: 'Lux Isla',
+        city: 'Ibiza',
+        country: 'Spain',
+    },
+    {
+        name: 'Nox Hostel',
+        city: 'London',
+        country: 'UK',
+    },
+    {
+        name: 'Leonardo Vienna',
+        city: 'Vienna',
+        country: 'Austria',
+    },
+    {
+        name: 'Adagio Aparthotel',
+        city: 'Edinburgh',
+        country: 'UK',
+    },
+    {
+        name: 'Steigenberger Hotel',
+        city: 'Hamburg',
+        country: 'Germany',
+    },
+];
+console.log(searchHotels('B', hotels));
+
+function getCity (hotels) {
+    let object = {};
+    for(let i=0; i<hotels.length; i++){
+        let currentPair = hotels[i];
+       if (currentPair.country in object) {
+           object[currentPair.country].push(currentPair.city)
+       } else {
+           object[currentPair.country] = [currentPair.city];
+       }
+    }
+   return object;
+}
+console.log(getCity(hotels));
+
+function getCalendarMonth(daysInMonth, daysInWeek, dayOfWeek) {
+    if (dayOfWeek >= daysInWeek)
+        throw Error(`${dayOfWeek} >= ${daysInWeek}`)
+
+    let month = []
+    let x = (daysInWeek - ((daysInMonth + dayOfWeek) % daysInWeek)) % daysInWeek // count of days to complete last week.
+    for (let i = 0; i < daysInMonth + dayOfWeek + x; i++) {
+        if (i % daysInWeek == 0) {
+            month.push([])
+        }
+        let last = month.length - 1
+        if (i < dayOfWeek) {
+            month[last].push(daysInMonth - dayOfWeek + i + 1)
+        } else if (i >= daysInMonth + dayOfWeek) {
+            month[last].push(i % daysInMonth - dayOfWeek + 1)
+        } else {
+            month[last].push(i - dayOfWeek + 1);
+        }
+    }
+    return month;
+}
+console.log(getCalendarMonth(30, 7, 4));
+
+/*lesson 7*/
+const obj1 = {
+    a: 'a',
+    b: {
+        a: 'a',
+        b: 'b',
+        c: {
+            a: 1,
+        },
+    },
+};
+const obj2 = {
+    b: {
+        c: {
+            a: 1,
+        },
+        b: 'b',
+        a: 'a',
+    },
+    a: 'a',
+};
+const obj3 = {
+    a: {
+        c: {
+            a: 'a',
+        },
+        b: 'b',
+        a: 'a',
+    },
+    b: 'b',
+};
+const deepEqual = (object1, object2) => {
```

Comment:
overall method looks too complex and hard to understand. could it be simplified?

Reply:
I can't make this function more simple for now